### PR TITLE
Firmware

### PR DESCRIPTION
**Pull request by MedvedEE**
Mainboard firmware works

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+firmware/Debug/
+t
+.gitignore
```
Comment (ReikoR) [inline .gitignore:3]: `t` is probably incorrect here and `.gitignore` should not be ignored.

File: firmware/USB_Device/App/usbd_cdc_if.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+/** @defgroup USBD_CDC_IF_Exported_Defines USBD_CDC_IF_Exported_Defines
+  * @brief Defines.
+  * @{
+  */
+/* Define size for the receive and transmit buffer over CDC */
+#define APP_RX_DATA_SIZE  1000
+#define APP_TX_DATA_SIZE  1000
+/* USER CODE BEGIN EXPORTED_DEFINES */
+ void CDC_On_Receive(uint8_t* buffer, uint32_t* length); // (1)
```
Comment (ReikoR) [inline firmware/USB_Device/App/usbd_cdc_if.h:54]: A bit more correct place for this function declaration would be at the end of the file between these lines:
```
/* USER CODE BEGIN EXPORTED_FUNCTIONS */

/* USER CODE END EXPORTED_FUNCTIONS */
```

**Review by ReikoR — CHANGES_REQUESTED** [review]
Some minor problems.

**Comment by MedvedEE** [comment]
Fixed

**Review by ReikoR — APPROVED** [review]

**Pull request merged**